test/03_device-state.ts

```diff
@@ -276,29 +281,38 @@ mockery.registerMock('../src/lib/config', configMock);
 								devicePollInterval / 1000
 							} seconds`, async () => {
 								stateChangeEventSpy.resetHistory();
+								let stateUpdatedAfter = new Date();
 								await setTimeout(devicePollInterval);
 
-								await waitFor({ checkFn: () => stateChangeEventSpy.called });
+								await waitFor({
+									checkFn: () => {
+										if (stateChangeEventSpy.called) {
+											return true;
+										}
+										stateUpdatedAfter = new Date();
+										return false;
+									},
+								});
 
 								expect(tracker.states[getDevice().id]).to.equal(
 									DeviceOnlineStates.Timeout,
 								);
 
-								const { body } = await supertest(getActor())
-									.get(`/${version}/device(${getDevice().id})`)
-									.expect(200);
-
-								expect(body.d[0]).to.not.be.undefined;
-								expect(body.d[0]).to.have.property(
-									'api_heartbeat_state',
-									DeviceOnlineStates.Timeout,
-									'API heartbeat state is not timeout',
+								await expectResourceToMatch(
+									getActor(),
+									'device',
+									getDevice().id,
+									{
+										api_heartbeat_state: DeviceOnlineStates.Timeout,
+										last_changed_api_heartbeat_state_on__date:
+											thatIsDateStringAfter(stateUpdatedAfter),
+									},
```

Comment:
When we use the `last_changed_api_heartbeat_state_on__date` what do we want to convey? Do we want to convey when the device was last seen with a proper heartbeat or when the heartbeat status changes from online, to timeout, to offline?
How will we use the timeout and offline timestamps, eg as indicator in the UI?